pallets/identity/src/lib.rs

```diff
@@ -1977,6 +1977,24 @@ impl<T: Trait> Module<T> {
 }
 
 impl<T: Trait> Module<T> {
+    /// RPC call to fetch some aggregate account data for fewer round trips.
+    pub fn get_key_identity_data(acc: T::AccountId) -> Option<types::KeyIdentityData<IdentityId>> {
+        let identity = Self::get_identity(&acc)?;
+        let record = <DidRecords<T>>::get(identity);
+        let permissions = if acc == record.primary_key {
+            None
+        } else {
+            Some(record.secondary_keys.into_iter().find_map(|sk| {
+                sk.signer.as_account().filter(|&a| a == &acc)?;
+                Some(sk.permissions)
+            })?)
+        };
+        Some(types::KeyIdentityData {
+            identity,
+            permissions,
+        })
```

Comment:
We need to differentiate between primary and secondary keys. Primary keys actually have all the permissions but this function will return None and there's no way to tell if the None represents a sk with no permission or a pk.